sync/src/relayer/mod.rs

```diff
@@ -305,17 +299,11 @@ impl<CS: ChainStore + 'static> Relayer<CS> {
 
         if !short_ids_set.is_empty() {
             let chain_state = self.shared().lock_chain_state();
-            for entry in chain_state.tx_pool().proposed_txs_iter() {
-                let short_id = short_transaction_id(key0, key1, &entry.transaction.witness_hash());
-                if short_ids_set.remove(&short_id) {
-                    txs_map.insert(short_id, entry.transaction.clone());
-
-                    // Early exit here for performance
-                    if short_ids_set.is_empty() {
-                        break;
-                    }
+            short_ids_set.into_iter().for_each(|short_id| {
+                if let Some(tx) = chain_state.tx_pool().get_tx(&short_id) {
```

Comment:
Add a new method for it? Because of the fetching order.

https://github.com/nervosnetwork/ckb/blob/develop/shared/src/tx_pool/pool.rs#L192-L200

Reply:
Updated, please review again, thanks.

Reply:
I am very confused about `get_tx` and `get_proposed_and_others`.
Please add some comments to explain the differences.
Why the fetching order matters and why we don't change the fetching order of `get_tx` and reuse it?

Reply:
Most txs in compact block will be in the proposal pool when we are trying to reconstruct, so add a new fn here to fetch tx from proposal pool first, then gap and pending pool.